sdk/go/common/workspace/plugins.go

```diff
@@ -239,25 +239,69 @@ func (info PluginInfo) Download() (io.ReadCloser, int64, error) {
 		return nil, -1, errors.Errorf("unknown version for plugin %s", info.Name)
 	}
 
-	// If the plugin has a server, associated with it, download from there.  Otherwise use the "default" location, which
-	// is hosted by Pulumi.
-	serverURL := info.PluginDownloadURL
-	if serverURL == "" {
-		serverURL = "https://get.pulumi.com/releases/plugins"
+	if info.PluginDownloadURL != "" {
+		return getPluginResponse(
+			buildUserSpecifiedPluginURL(info.PluginDownloadURL, info.Kind, info.Name, info.Version, opSy, arch))
 	}
-	serverURL = interpolateURL(serverURL, *info.Version, os, arch)
+
+	if _, ok := os.LookupEnv("PULUMI_EXPERIMENTAL"); ok {
+		pluginURL := buildGitHubReleasesPluginURL(info.Kind, info.Name, info.Version, opSy, arch)
+
+		resp, length, err := getPluginResponse(pluginURL)
+		if err == nil {
+			return resp, length, nil
+		}
+
+		// we threw an error talking to GitHub so lets fallback to get.pulumi.com for the provider
+		logging.V(1).Infof("cannot find plugin on github.com/pulumi/pulumi-%s/releases", info.Name)
+	}
+
+	return getPluginResponse(buildPulumiHostedPluginURL(info.Kind, info.Name, info.Version, opSy, arch))
+}
+
+func buildGitHubReleasesPluginURL(kind PluginKind, name string, version *semver.Version, opSy, arch string) string {
+	logging.V(1).Infof("%s downloading from github.com/pulumi/pulumi-%s/releases", name, name)
+
+	return fmt.Sprintf("https://github.com/pulumi/pulumi-%s/releases/download/v%s/%s",
+		name, version.String(), url.QueryEscape(fmt.Sprintf("pulumi-%s-%s-v%s-%s-%s.tar.gz",
+			kind, name, version.String(), opSy, arch)))
+}
+
+func buildPulumiHostedPluginURL(kind PluginKind, name string, version *semver.Version, opSy, arch string) string {
```

Comment:
This method is just `buildUserSpecifiedPluginURL("https://get.pulumi.com/releases/plugins", kind, name, version, opSy, arch)`